Description: Enable ZFS support in grub-probe and grub-setup
Author: Robert Millan <rmh@debian.org>
Forwarded: not-needed
Last-Update: 2010-09-10

--- a/conf/common.rmk	2010-07-06 18:27:55 +0000
+++ b/conf/common.rmk	2010-07-29 14:19:10 +0000
@@ -35,11 +35,17 @@
 	fs/sfs.c fs/ufs.c fs/ufs2.c fs/xfs.c fs/afs.c		\
 	fs/afs_be.c fs/befs.c fs/befs_be.c fs/tar.c		\
 	\
+	$(GRUB_CONTRIB)/zfs/zfs.c \
+	$(GRUB_CONTRIB)/zfs/zfs_lzjb.c \
+	$(GRUB_CONTRIB)/zfs/zfs_sha256.c \
+	$(GRUB_CONTRIB)/zfs/zfs_fletcher.c \
+	\
 	partmap/msdos.c partmap/bsdlabel.c partmap/apple.c \
 	partmap/sun.c partmap/sunpc.c partmap/gpt.c \
 	kern/fs.c kern/env.c fs/fshelp.c			\
 	disk/raid.c disk/raid5_recover.c disk/raid6_recover.c	\
 	disk/mdraid_linux.c disk/lvm.c grub_probe_init.c
+grub_probe_CFLAGS = -I$(GRUB_CONTRIB)/zfs/include
 
 ifeq ($(enable_grub_fstest), yes)
 bin_UTILITIES += grub-fstest
@@ -60,6 +66,11 @@
 	fs/ufs.c fs/ufs2.c fs/xfs.c fs/afs.c fs/afs_be.c fs/befs.c 	\
 	fs/befs_be.c fs/tar.c			\
 	\
+	$(GRUB_CONTRIB)/zfs/zfs.c \
+	$(GRUB_CONTRIB)/zfs/zfs_lzjb.c \
+	$(GRUB_CONTRIB)/zfs/zfs_sha256.c \
+	$(GRUB_CONTRIB)/zfs/zfs_fletcher.c \
+	\
 	kern/partition.c partmap/msdos.c partmap/bsdlabel.c 		\
 	partmap/apple.c partmap/sun.c partmap/sunpc.c partmap/gpt.c	\
 	kern/fs.c kern/env.c fs/fshelp.c disk/raid.c			\
 	disk/raid5_recover.c disk/raid6_recover.c 			\
 	disk/mdraid_linux.c disk/dmraid_nvidia.c disk/lvm.c 		\
 	grub_fstest_init.c
+grub_fstest_CFLAGS = -I$(GRUB_CONTRIB)/zfs/include
 
 # For grub-mkfont.
 ifeq ($(enable_grub_mkfont), yes)
--- a/conf/i386-pc.rmk	2010-07-06 18:27:55 +0000
+++ b/conf/i386-pc.rmk	2010-07-29 14:19:31 +0000
@@ -75,6 +75,12 @@
 	fs/sfs.c fs/ufs.c fs/ufs2.c fs/xfs.c fs/afs.c			\
 	fs/afs_be.c fs/befs.c fs/befs_be.c fs/tar.c			\
 									\
+	$(GRUB_CONTRIB)/zfs/zfs.c \
+	$(GRUB_CONTRIB)/zfs/zfs_lzjb.c \
+	$(GRUB_CONTRIB)/zfs/zfs_sha256.c \
+	$(GRUB_CONTRIB)/zfs/zfs_fletcher.c \
+									\
+									\
 	partmap/msdos.c partmap/bsdlabel.c partmap/sunpc.c		\
 	partmap/gpt.c							\
 									\
@@ -82,6 +88,7 @@
 	disk/mdraid_linux.c disk/lvm.c					\
 	util/raid.c util/lvm.c						\
 	grub_setup_init.c
+grub_setup_CFLAGS = -I$(GRUB_CONTRIB)/zfs/include
 
 sbin_SCRIPTS += grub-install
 grub_install_SOURCES = util/grub-install.in

--- a/conf/sparc64-ieee1275.rmk	2010-07-06 18:27:55 +0000
+++ b/conf/sparc64-ieee1275.rmk	2010-07-29 14:19:40 +0000
@@ -59,12 +59,18 @@
 	fs/sfs.c fs/ufs.c fs/ufs2.c fs/xfs.c fs/afs.c			\
 	fs/afs_be.c fs/befs.c fs/befs_be.c fs/tar.c			\
 									\
+	$(GRUB_CONTRIB)/zfs/zfs.c \
+	$(GRUB_CONTRIB)/zfs/zfs_lzjb.c \
+	$(GRUB_CONTRIB)/zfs/zfs_sha256.c \
+	$(GRUB_CONTRIB)/zfs/zfs_fletcher.c \
+									\
 	partmap/amiga.c	partmap/apple.c partmap/msdos.c			\
 	partmap/bsdlabel.c partmap/sun.c partmap/acorn.c		\
 									\
 	disk/raid.c disk/raid5_recover.c disk/raid6_recover.c		\
 	disk/mdraid_linux.c disk/lvm.c util/raid.c			\
 	util/lvm.c gnulib/progname.c grub_setup_init.c
+grub_setup_CFLAGS = -I$(GRUB_CONTRIB)/zfs/include
 
 # For grub-ofpathname.
 grub_ofpathname_SOURCES = util/ieee1275/grub-ofpathname.c	\

